volatility3/framework/symbols/linux/extensions/__init__.py

```diff
@@ -539,3 +545,248 @@ def reference_count(self):
+        protocol = IP_PROTOCOLS.get(self.sk.sk_protocol, "UNKNOWN")
+        if self.family == "AF_INET6":
+            protocol = IPV6_PROTOCOLS.get(self.sk.sk_protocol, protocol)
+        return protocol
+
+    @property
+    def state(self):
+        """Return a string representing the sock state."""
+
+        if self.sk.type == "STREAM":
+            state_idx = self.sk.__sk_common.skc_state
+            if 0 <= state_idx < len(TCP_STATES):
+                state = TCP_STATES[state_idx]
+            else:
+                state = "UNKNOWN"
+        else:
+            state = "UNCONNECTED"
+
+        return state
+
+    @property
+    def src_port(self):
+        sport_le = getattr(self, "sport", getattr(self, "inet_sport", None))
+        if sport_le is not None:
+            return socket.htons(sport_le)
+
+    @property
+    def dst_port(self):
+        sk_common = self.sk.__sk_common
+        if hasattr(sk_common, "skc_portpair"):
+            dport_le = sk_common.skc_portpair & 0xffff
+        elif hasattr(self, "dport"):
+            dport_le = self.dport
+        elif hasattr(self, "inet_dport"):
+            dport_le = self.inet_dport
+        elif hasattr(sk_common, "skc_dport"):
+            dport_le = sk_common.skc_dport
+        else:
+            return
+
+        return socket.htons(dport_le)
+
+    @property
+    def src_addr(self):
+        sk_common = self.sk.__sk_common
+        family = sk_common.skc_family
+        if family == socket.AF_INET:
+            addr_size = 4
+            if hasattr(self, "rcv_saddr"):
+                saddr = self.rcv_saddr
+            elif hasattr(self, "inet_rcv_saddr"):
+                saddr = self.inet_rcv_saddr
+            else:
+                saddr = sk_common.skc_rcv_saddr
+        elif family == socket.AF_INET6:
+            addr_size = 16
+            saddr = self.pinet6.saddr
+        else:
+            return
+
+        parent_layer = self._context.layers[self.vol.layer_name]
+        addr_bytes = parent_layer.read(saddr.vol.offset, addr_size)
+        return socket.inet_ntop(family, addr_bytes)
+
+    @property
+    def dst_addr(self):
+        sk_common = self.sk.__sk_common
+        family = sk_common.skc_family
+        if family == socket.AF_INET:
+            if hasattr(self, "daddr") and self.daddr:
+                daddr = self.daddr
+            elif hasattr(self, "inet_daddr") and self.inet_daddr:
+                daddr = self.inet_daddr
+            else:
+                daddr = sk_common.skc_daddr
+            addr_size = 4
+        elif family == socket.AF_INET6:
+            if hasattr(self.pinet6, "daddr"):
+                daddr = self.pinet6.daddr
+            else:
+                daddr = sk_common.skc_v6_daddr
+            addr_size = 16
+        else:
+            return
+
+        parent_layer = self._context.layers[self.vol.layer_name]
+        addr_bytes = parent_layer.read(daddr.vol.offset, addr_size)
```

Comment:
Also needs a try/except for invalid addresses.

Reply:
same question as above